users/api/http/requests.go

```diff
@@ -27,3 +27,53 @@ func (req viewUserInfoReq) validate() error {
 	}
 	return nil
 }
+
+type passwResetReq struct {
+	Email string `json:"email"`
+	Host  string `json:"host,omitempty"`
+}
+
+func (req passwResetReq) validate() error {
+	if req.Email == "" || req.Host == "" {
+		return users.ErrMalformedEntity
+	}
+	return nil
+}
+
+type resetTokenReq struct {
+	Token    string `json:"token,omitempty"`
+	Password string `json:"password,omitempty"`
+	ConfPass string `json:"confirmPassword,omitempty"`
+}
+
+func (req resetTokenReq) validate() error {
+	if req.Password == "" || req.ConfPass == "" {
+		return users.ErrMalformedEntity
+	}
+	if req.Token == "" {
+		return users.ErrMissingResetToken
+	}
+	if req.Password != req.ConfPass {
+		return users.ErrMalformedEntity
+	}
+	return nil
+}
+
+type passwChangeReq struct {
+	Token       string `json:"token,omitempty"`
+	Password    string `json:"password,omitempty"`
+	OldPassword string `json:"oldPassword,omitempty"`
```

Comment:
Just to check, do you need all of these `omitempty`? The same question goes for the other request.